doc/contribute/guidelines.rst

```diff
@@ -554,6 +570,31 @@ exceptions:
 * Avoid using binary literals (constants starting with ``0b``).
 * Avoid using non-ASCII symbols in code, unless it significantly improves
   clarity, avoid emojis in any case.
+* Use proper capitalization of nouns in code comments (e.g. ``UART`` and not
+  ``uart``, ``CMake`` and not ``cmake``).
+
+Beyond C code, the following coding style rules apply to other types of files:
+
+* CMake
+
+  * Indent with spaces, indentation is two spaces.
+  * No space between commands (e.g. ``if``) and the corresponding opening
+    bracket (e.g. ``(``).
+
+* Devicetree
+
+  * Indent with tabs.
+  * No empty lines before a dedenting ``};``.
+  * Single empty line separating nodes at the same hierarchy level.
+
+* Kconfig
```

Comment:
There's a few additional style items listed at
https://builds.zephyrproject.io/zephyr/pr/83117/docs/build/kconfig/tips.html#header-comments-and-other-nits

We should have one location for the Kconfig style and just have the other locations link to the source of truth.

Reply:
I will remove that from the Kconfig guide and instead move it here, where the coding style is actually defined.

Reply:
If you do remove them from the Kconfig I suggest you at least add a link there to reference back here.